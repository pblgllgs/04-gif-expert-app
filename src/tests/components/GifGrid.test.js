import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
//finje la llamada a ese fetch
jest.mock("../../hooks/useFetchGifs");

describe("Prueba en <GifGrid />", () => {
    const category = "HunterX";

    test("Debe mostrar <GifGrid /> correctamente", () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test("debe de mostrar items cuando se cargan imagenes con el useFetchGifs", () => {
        const gifs = [
            {
                id: "abc",
                title: "titulo1",
                url: "http://foto.imagen/photo1.jpg",
            },
            {
                id: "123",
                title: "titulo2",
                url: "http://foto.imagen/photo2.jpg",
            }
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });


});
