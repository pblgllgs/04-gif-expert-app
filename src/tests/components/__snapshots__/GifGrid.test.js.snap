// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGrid /> Debe mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    HunterX
  </h3>
  <p
    className="animate__animated animate__flash animate__slower	3s"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
